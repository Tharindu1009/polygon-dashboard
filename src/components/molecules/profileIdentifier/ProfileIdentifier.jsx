import './styles/profileIdentifier.scss';
import * as React from 'react';
import User from '../../../assets/images/user.png';

function ProfileIdentifier({name, isDisplayName}) {
    return (
        <div className='profileIdentifier' data-testid="profileIdentifier">
            {isDisplayName && <div className='profile-name'>{name}</div>}
            <img className='image' src={User} alt='Profile' />
        </div>
    );
}

export default ProfileIdentifier;
