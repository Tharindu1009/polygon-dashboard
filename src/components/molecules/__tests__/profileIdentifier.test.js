import { render, screen, cleanup } from "@testing-library/react";
import TestRenderer from 'react-test-renderer';
import ProfileIdentifier from "../profileIdentifier/ProfileIdentifier";

afterEach(() => {
    cleanup();
})

test('should render ProfileIdentifier molecule', () => {
    render(<ProfileIdentifier name="user1009" isDisplayName={true} />);
    const profileIdentifier = screen.getByTestId('profileIdentifier');
    expect(profileIdentifier).toBeInTheDocument();
    expect(profileIdentifier).toHaveTextContent("user1009");
})

test('matches ProfileIdentifier snapshot', () => {
    const tree = TestRenderer.create(<ProfileIdentifier name="user1009" isDisplayName={true} />).toJSON();
    expect(tree).toMatchSnapshot();
})