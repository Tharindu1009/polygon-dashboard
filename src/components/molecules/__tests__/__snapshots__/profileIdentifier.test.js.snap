// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches ProfileIdentifier snapshot 1`] = `
<div
  className="profileIdentifier"
  data-testid="profileIdentifier"
>
  <div
    className="profile-name"
  >
    user1009
  </div>
  <img
    alt="Profile"
    className="image"
    src="user.png"
  />
</div>
`;
